File: 279_div2c.cpp
#include <bits/stdc++.h>
using namespace std ;
int main ()
{
	string s;
	long long int length,l;
	cin>>s;
	length=s.length();
	long long int p,a,b,num,rem1[length],rem2[length];
	cin>>a>>b;
	num=0;
	for(long long int i=0;i<length;i++)
	{
		num=((num*10)%a+(s[i]-'0'))%a;
		rem1[i]=num%a;
	}
	p=1;
	num=0;
	bool flag1= 0 ;
	for(long long int i=length-1;i>=0;i--)
	{
		num=(num+(p*(s[i]-'0')%b)%b)%b;
		p=(p*10)%b;
		rem2[i]=num;
	}
	for(long long int i=0;i<length-1;i++)
	{
		if(rem1[i]==0 && rem2[i+1]==0 && s[i+1]!='0')
		{
			cout << "YES\n";
			cout << s.substr(0,i+1) << "\n" ;
			cout << s.substr(i+1,length);
			flag1=1 ;
			break ;
		}
	}
	if(flag1==0)
		cout<<"NO";
	return 0;
}

File: 279_div2c.py
s = raw_input()
a,b = map(int,raw_input().split())
flag1=0
for x in xrange(1,len(str(s))):
	s2=s[x:len(str(s))]
	s1=int(s)-int(s2)
	if int(str(s1)[0])!=0 and int(str(s2)[0])!=0:
		if int(str(s1)[0:x])%a==0:
			if int(s2)%b==0:
			#print s1,s2
				flag1=1
				print "YES"
				print str(s1)[0:x]
				print s2
				break
if flag1==0:
	print "NO"